### PR TITLE
[QOL-7857] install all extensions on batch nodes in case they alter batch processing

### PR DESCRIPTION
**Pull request by ThrawnCA**
Open Data DEV is getting errors because running XLoader wipes the extra fields that ckanext-scheming expects.

### CODE REVIEW

File: recipes/ckanweb-deploy-exts.rb [hunk 1]
```diff
@@ -117,305 +115,300 @@
 	apprevision ||= app['app_source']['url'][/@(.*)/].sub '@', ''
 	apprevision ||= "master"
 
-	# Don't install extensions not required by the batch node
+	apprelease = app['app_source']['url'].sub('http', "git+http").sub(/@(.*)/, '')
```
Comment (duttonw) [inline recipes/ckanweb-deploy-exts.rb:118]: I've seen you use include scripts for opsworks/chef, could this be pulled out or not required if it is also being run on the ckan-setup section?

**Review by duttonw — COMMENTED** [review]
like a split on the extension installation as it looks the same as ckan web (or moved to ckan-setup part of the scripts)

File: recipes/ckanweb-deploy-exts.rb (hunk 1 above)
Comment (ThrawnCA) [reply recipes/ckanweb-deploy-exts.rb:118]: This is being run for each extension app. It can't just reuse the value from the CKAN core app.

Comment (duttonw) [reply recipes/ckanweb-deploy-exts.rb:118]: was not meaning just this but the entire extension installation process is on both ckan web and ckan batch

Comment (ThrawnCA) [reply recipes/ckanweb-deploy-exts.rb:118]: Offhand I'm not sure how to define a parameterised recipe. I would use that in several places in our cookbook if I knew an easy way.

Comment (ThrawnCA) [reply recipes/ckanweb-deploy-exts.rb:118]: Okay, https://docs.chef.io/custom_resources/ looks promising. Do you want me to look into applying that before we merge, or I can raise a separate pull request afterward?

Comment (duttonw) [reply recipes/ckanweb-deploy-exts.rb:118]: go seperate

**Review by duttonw — APPROVED** [review]

**Pull request merged**